src/test/java/org/folio/circulation/resources/RequestQueueResourceTest.java

```diff
@@ -0,0 +1,315 @@
+      is("There is inconsistency between provided reordered queue and item queue."));
+  }
+
+  @Test
+  public void validationErrorWhenNotAllRequestsProvidedInReorderedQueue() throws Exception {
+    loansFixture.checkOutByBarcode(item, usersFixture.rebecca());
+
+    holdRequest(steve);
+    IndividualResource secondRecallRequest = recallRequest(jessica);
+
+    Response response = requestQueueFixture.attemptReorderQueue(
+      item.getId().toString(),
+      new ReorderQueueBuilder()
+        .addReorderRequest(secondRecallRequest.getId().toString(), 1)
+        .create()
+    );
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody(),
+      is("There is inconsistency between provided reordered queue and item queue."));
+  }
+
+  @Test
+  public void canNotSaveRequestsWithDuplicatedPositions() throws Exception {
+    loansFixture.checkOutByBarcode(item, usersFixture.rebecca());
+
+    IndividualResource holdRequest = holdRequest(steve);
+    IndividualResource recallRequest = recallRequest(jessica);
+
+    Response response = requestQueueFixture.attemptReorderQueue(
+      item.getId().toString(),
+      new ReorderQueueBuilder()
+        .addReorderRequest(recallRequest.getId().toString(), 1)
+        .addReorderRequest(holdRequest.getId().toString(), 1)
+        .create()
+    );
+
+    assertThat(response.getStatusCode(), is(500));
+    assertThat(response.getBody(),
+      is("Some requests have not been updated. Fetch the queue and try again."));
+  }
+
+  @Test
+  @Parameters({
+    "1, 2, 3, 4",
+    "1, 2, 4, 3",
+    "2, 1, 3, 4",
+    "2, 1, 4, 3",
+    "4, 3, 2, 1",
+    "4, 3, 1, 2",
+    "3, 4, 2, 1",
+    "3, 4, 1, 2",
+    "3, 4, 1, 50",
+    "40, 2, 1, 3",
+  })
+  public void shouldReorderQueueSuccessfully(int firstPosition, int secondPosition,
+                                             int thirdPosition, int fourthPosition) throws Exception {
+    loansFixture.checkOutByBarcode(item, rebecca);
+
+    IndividualResource firstHoldRequest = holdRequest(steve);
+    IndividualResource secondHoldRequest = holdRequest(james);
+    IndividualResource firstRecallRequest = recallRequest(charlotte);
+    IndividualResource secondRecallRequest = recallRequest(jessica);
+
+    JsonObject reorderQueue = new ReorderQueueBuilder()
+      .addReorderRequest(firstHoldRequest.getId().toString(), firstPosition)
+      .addReorderRequest(secondHoldRequest.getId().toString(), secondPosition)
+      .addReorderRequest(firstRecallRequest.getId().toString(), thirdPosition)
+      .addReorderRequest(secondRecallRequest.getId().toString(), fourthPosition)
+      .create();
+
+    JsonObject response = requestQueueFixture
+      .reorderQueue(item.getId().toString(), reorderQueue);
+
+    assertQueue(reorderQueue, response);
+  }
+
+  @Test
+  @Parameters(source = ReorderQueueTestDataSource.class)
+  public void canReorderQueueTwice(Integer[] initialState, Integer[] targetState) throws Exception {
```

Comment:
What does this test demonstrate? That a queue can be reordered twice in succession?

Reply:
that queue can be reordered from a random initial position (different from 1, 2, 3, 4)